src/pool/srv_target.c

```diff
@@ -1075,22 +1075,49 @@ cont_iter_cb(uuid_t cont_uuid, vos_iter_entry_t *ent, void *data)
 	return arg->callback(cont_uuid, ent, arg->arg);
 }
 
+struct dtx_resync_args {
+	daos_handle_t	ph;
+	uuid_t		po_uuid;
+	uuid_t		co_uuid;
+	uint32_t	ver;
+};
+
+static int
+dtx_resync_ult(void *data)
+{
+	struct dtx_resync_args	*args = data;
+	int			 rc;
+
+	rc = dtx_resync(args->ph, args->po_uuid, args->co_uuid, args->ver,
+			true);
+	if (rc != 0)
+		D_ERROR("Fail to resync some DTX(s) for the pool/cont "
+			DF_UUID"/"DF_UUID" that will affect subsequent "
+			"object rebuild: rc = %d.\n",
+			DP_UUID(args->po_uuid), DP_UUID(args->co_uuid), rc);
+	return rc;
+}
+
 static int
 pool_iter_cb(daos_handle_t ph, uuid_t co_uuid, void *data)
 {
 	struct cont_iter_arg *arg = data;
 
 	switch (arg->intent) {
 	case DAOS_INTENT_REBUILD: {
-		int	rc;
+		struct dtx_resync_args	args;
+		int			rc;
 
 		/* For rebuild case, we need to resync DTXs' status firstly. */
-		rc = dtx_resync(ph, arg->po_uuid, co_uuid, arg->version, true);
+		args.ph = ph;
+		args.ver = arg->version;
+		uuid_copy(args.po_uuid, arg->po_uuid);
+		uuid_copy(args.co_uuid, co_uuid);
+		rc = dss_ult_create_execute(dtx_resync_ult, &args, NULL, NULL,
```

Comment:
This code needs be revised later since dtx_resync() yield in a iteration callback, and current container iteration code doesn't re-probe after yield. Same issue exists in current rebuild code.

I suppose Jeff will enhance the vos_iterate() to support VOS_ITER_COUUID, so that vos_iterate() can be used for rebuild then.

Reply:
agreed, the original code also with this potential issue. 
we can refine it later, or do you want to add a reprobe it here or add a TODO comment?

Reply:
No, I don't think we need to handle this in this patch.